controlplane/kubeadm/controllers/upgrade.go

```diff
@@ -82,44 +79,14 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(
 		return ctrl.Result{}, errors.Wrap(err, "failed to upgrade kubelet config map")
 	}
 
-	// If there is not already a Machine that is marked for upgrade, find one and mark it
-	selectedForUpgrade := requireUpgrade.Filter(machinefilters.HasAnnotationKey(controlplanev1.SelectedForUpgradeAnnotation))
-	if len(selectedForUpgrade) == 0 {
-		selectedMachine, err := r.selectAndMarkMachine(ctx, requireUpgrade, controlplanev1.SelectedForUpgradeAnnotation, controlPlane)
-		if err != nil {
-			logger.Error(err, "failed to select machine for upgrade")
-			return ctrl.Result{}, err
-		}
-		selectedForUpgrade = selectedForUpgrade.Insert(selectedMachine)
-	}
-
-	replacementCreated := selectedForUpgrade.Filter(machinefilters.HasAnnotationKey(controlplanev1.UpgradeReplacementCreatedAnnotation))
-	if len(replacementCreated) == 0 {
-		// TODO: should we also add a check here to ensure that current machines not > kcp.spec.replicas+1?
-		// We haven't created a replacement machine for the cluster yet
-		// return here to avoid blocking while waiting for the new control plane Machine to come up
-		result, err := r.scaleUpControlPlane(ctx, cluster, kcp, ownedMachines, controlPlane)
-		if err != nil {
-			return ctrl.Result{}, err
-		}
-		if err := r.markWithAnnotationKey(ctx, selectedForUpgrade.Oldest(), controlplanev1.UpgradeReplacementCreatedAnnotation); err != nil {
-			return ctrl.Result{}, err
-		}
-		return result, nil
-	}
-
-	return r.scaleDownControlPlane(ctx, cluster, kcp, ownedMachines, replacementCreated, controlPlane)
-}
-
-func (r *KubeadmControlPlaneReconciler) selectAndMarkMachine(ctx context.Context, machines internal.FilterableMachineCollection, annotation string, controlPlane *internal.ControlPlane) (*clusterv1.Machine, error) {
-	selected, err := controlPlane.MachineInFailureDomainWithMostMachines(machines)
+	status, err := workloadCluster.ClusterStatus(ctx)
```

Comment:
I know this is not part of this PR, but ClusterStatus confused me.
It is ControlPlane status (workers are not considered)...

Reply:
Oh, nevermind! I thought your point was that I am looking at all nodes, and not just control plane nodes, and this was a huge bug 🤦 . But in fact it's that this method *is* only for control plane nodes, even though it seems more like it should be for all nodes.